lib/elixir/lib/access.ex

```diff
@@ -360,6 +382,11 @@ defmodule Access do
   end
 
   def pop(nil, key) do
+    pop_raise(key)
+  end
+
+  @doc false
+  def pop_raise(key) do
```

Comment:
Let's go with the previous approach you suggested and copy the error message to Kernel. Thank you!